homeassistant/components/litterrobot/switch.py

```diff
@@ -0,0 +1,89 @@
+"""Support for Litter-Robot switches."""
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .hub import LitterRobotEntity
+
+NIGHT_LIGHT = "Night Light"
+PANEL_LOCKOUT = "Panel Lockout"
+SLEEP_MODE = "Sleep Mode"
+
+DEFAULT_SLEEP_TIME = "22:00"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Litter-Robot switches using config entry."""
+    hub = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for robot in hub.account.robots:
+        entities.append(LitterRobotSwitch(robot, NIGHT_LIGHT, hub))
+        entities.append(LitterRobotSwitch(robot, PANEL_LOCKOUT, hub))
+        entities.append(LitterRobotSwitch(robot, SLEEP_MODE, hub))
+
+    if entities:
+        async_add_entities(entities, True)
+
+
+class LitterRobotSwitch(LitterRobotEntity, ToggleEntity):
+    """Litter-Robot Switches."""
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        if self.entity_type == NIGHT_LIGHT:
```

Comment:
This would be cleaner to make 3 class of entities since there isn't any shared functionality.